File: Single/Easy/Temperatures.java
// Temperatures
import java.util.*;

class Solution {
    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        int N = in.nextInt();           // the number of temperatures to analyse
        in.nextLine();
        String TEMPS = in.nextLine();   // the N temperatures expressed as integers ranging from -273 to 5526
        int result = 10000;
        int res = 0;
        int t = 0;
        if (N == 0) {
            System.out.println(0);
        } else {
            String[] temperatures = TEMPS.trim().split(" ");
            int[] temps = new int[temperatures.length];
            for (int i = 0; i < temps.length; ++i) {
                temps[i] = Integer.parseInt(temperatures[i]);
                System.err.println(temps[i]);
            }
            for (int i = 0; i < temps.length; ++i) {
                res = Math.abs(result);
                t = Math.abs(temps[i]);
                if (res == t) {
                    if (result < temps[i]) { result = temps[i]; }    
                } else if (res > t) {
                    result = temps[i];
                }
            }
            System.out.println(result);
        }
    }
}

File: Single/Easy/Temperatures.php
<?php
// Temperatures
    fscanf(STDIN, "%d", $N);                    // Number of temperatures to analyse
    $TEMPS = stream_get_line(STDIN, 256, "\n"); // N temperatures expressed as integers ranging: -273 to 5526
    $result = 10000;
    $res = $t = 0;
    if ($N == 0) {
        print("0\n");
    } else {
        $temperatures = explode(" ", trim($TEMPS));
        $temps = array();
        for ($i = 0; $i < count($temperatures); $i++) {
            $temps[$i] = (integer)$temperatures[$i];
            error_log($temps[$i]);
        }
        for ($i = 0; $i < count($temps); $i++) {
            $res = abs($result);
            $t = abs($temps[$i]);
            if ($res == $t) {
                if ($result < $temps[$i]) { $result = $temps[$i]; }    
            } else if ($res > $t) {
                $result = $temps[$i];
            }
        }
        print("$result\n");
    }
?>